packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js

```diff
@@ -238,6 +244,62 @@ const ApiTokenCreateView = () => {
                           </Textarea>
                         </GridItem>
+                        <GridItem key="duration" col={6} xs={12}>
+                          <Select
+                            name="duration"
+                            label={formatMessage({
+                              id: 'Settings.apiTokens.form.duration',
+                              defaultMessage: 'Token duration',
+                            })}
+                            value={values.duration}
+                            error={
+                              errors.duration
+                                ? formatMessage(
+                                    errors.duration?.id
+                                      ? errors.duration
+                                      : { id: errors.duration, defaultMessage: errors.duration }
+                                  )
+                                : null
+                            }
+                            onChange={value => {
+                              handleChange({ target: { name: 'duration', value } });
+                            }}
+                            required
+                            disabled={!isCreating}
+                          >
+                            <Option value="7">
+                              {formatMessage({
+                                id: 'Settings.apiTokens.duration.7-days',
+                                defaultMessage: '7 days',
+                              })}
+                            </Option>
+                            <Option value="30">
+                              {formatMessage({
+                                id: 'Settings.apiTokens.duration.30-days',
+                                defaultMessage: '30 days',
+                              })}
+                            </Option>
+                            <Option value="90">
+                              {formatMessage({
+                                id: 'Settings.apiTokens.duration.90-days',
+                                defaultMessage: '90 days',
+                              })}
+                            </Option>
+                            <Option value="unlimited">
+                              {formatMessage({
+                                id: 'Settings.apiTokens.duration.unlimited',
+                                defaultMessage: 'Unlimited',
+                              })}
+                            </Option>
+                          </Select>
+                          <Typography variant="pi" textColor="neutral600">
+                            {!isCreating &&
+                              `${formatMessage({
+                                id: 'Settings.apiTokens.duration.expiration-date',
+                                defaultMessage: 'Expiration date',
+                              })}: ${getDateOfExpiration(apiToken?.createdAt, values.duration)}`}
+                          </Typography>
```

Comment:
small translation improvement possible here I think 🤔
the date will never be localized here: 
<img width="619" alt="Screenshot 2022-07-04 at 13 36 46" src="https://user-images.githubusercontent.com/71838159/177148298-c66d47d9-c05c-42d9-979a-b6fa0ecac0b6.png">

I checked the `date-fns` [documentation ](https://date-fns.org/v2.28.0/docs/I18n)and it seems like we could translate it by passing a third argument to `format`, this way:

```
import * as locales from 'date-fns/locale';

const getDateOfExpiration = (createdAt, duration) =>
  format(addDays(new Date(createdAt), duration), 'PPP', { locale: locales?.fr });
```
result: 
<img width="992" alt="image" src="https://user-images.githubusercontent.com/71838159/177148952-81aedd3c-83a0-4beb-853c-3c6d44b73889.png">

I was wondering how to get the current locale from the user and actually we have it in the localStorage:
<img width="307" alt="Screenshot 2022-07-04 at 13 42 46" src="https://user-images.githubusercontent.com/71838159/177148843-03bd749a-5341-4c6f-96b6-901dffe0a8cd.png">

WDYT?

Reply:
oh yes, didn't think about the translation of the date itself, it's nice that they have an easy way to implement this, yeah I will implement your Idea.
and yeah we didn't have any tests implemented for this page I guess I can add tests